pkg/plugin/aqua/plugin.go

```diff
@@ -129,59 +261,156 @@ func (s *plugin) newScanJobContainer(ctx starboard.PluginContext, podContainer c
 				Name:  "AQUA_VERSION",
 				Value: version,
 			},
+			constructEnvVarSourceFromConfigMap("AQUA_CSP_HOST", pluginConfigName, keyAquaCspHost),
+			constructEnvVarSourceFromConfigMap("AQUA_CSP_USERNAME", pluginConfigName, keyAquaUsername),
+			constructEnvVarSourceFromConfigMap("AQUA_CSP_PASSWORD", pluginConfigName, keyAquaPassword),
+		},
+		Resources: requirements,
+		VolumeMounts: []corev1.VolumeMount{
+			{
+				Name:      "scannercli",
+				MountPath: "/usr/local/bin/scannercli",
+				SubPath:   "scannercli",
+			},
 			{
-				Name: "AQUA_CSP_HOST",
-				ValueFrom: &corev1.EnvVarSource{
-					ConfigMapKeyRef: &corev1.ConfigMapKeySelector{
-						LocalObjectReference: corev1.LocalObjectReference{
-							Name: starboard.GetPluginConfigMapName("Aqua"),
+				Name:      "dockersock",
+				MountPath: "/var/run/docker.sock",
+			},
+		},
+	}, nil
+}
+
+const (
+	FsSharedVolumeName = "starboard-aqua"
+)
+
+func (s *plugin) getPodSpecForFileSystemCommand(ctx starboard.PluginContext, config Config, object client.Object) (corev1.PodSpec, []*corev1.Secret, error) {
+	spec, err := kube.GetPodSpec(object)
+	if err != nil {
+		return corev1.PodSpec{}, nil, err
+	}
+
+	initContainerName := s.idGenerator.GenerateID()
+
+	aquaImageRef, err := s.getImageRef(ctx)
+	if err != nil {
+		return corev1.PodSpec{}, nil, err
+	}
+	starboardAquaImage, err := config.GetStarboardAquaScannerImage()
+	if err != nil {
+		return corev1.PodSpec{}, nil, err
+	}
+	secretName := vulnerabilityreport.GetScanJobName(object) + "-volume"
+	var env []corev1.EnvVar
```

Comment:
the env param below is not used in `getPodSpecForFileSystemCommand` method 
```
var env []corev1.EnvVar
	envVars, err := s.getEnvFromConfig(ctx, secretName)
	if err != nil {
		return corev1.PodSpec{}, nil, err
	}
	env = append(env, envVars...)```

maybe it should be moved to newScanJobContainerFSCommand , it will reduce the amount of env var passed

Reply:
Thanks @chen-keinan , we will need to send secretName. but secretName will be less data as compared to slice of envVars. I will update PR.

Reply:
But I realized that we will need to ctx as well and env slice will be generated equal to the number of containers present in the original workload. So should we do that or leave it as current way.

Reply:
I see , then better keeping it the current way